pages/embedded-portal.tsx

```diff
@@ -264,16 +274,83 @@ export const getServerSideProps: GetServerSideProps = async (context) => {
     },
   });
 
+  if (!existingSpace) {
+    console.log("No space");
+    return {
+      props: {
+        environmentToken,
+        lastSyncedAt: lastSync
+          ? DateTime.fromJSDate(lastSync.createdAt).toFormat(
+              "MM/dd/yy hh:mm:ss a"
+            )
+          : "",
+        existingSpace,
+        userId: token.sub,
+      },
+    };
+  }
+
+  let spaceData = await getSpace({
+    workflow: WorkflowType.Embedded,
+    spaceId: (existingSpace?.flatfileData as unknown as FlatfileSpaceData).id,
+  });
+
+  console.log("spaceData", spaceData);
+
+  let workbook = await getWorkbook({
+    workflow: WorkflowType.Embedded,
+    workbookId: spaceData?.primaryWorkbookId!,
+  });
+
+  console.log("workbook", workbook);
+
+  const timeInFive = new Date();
+  timeInFive.setSeconds(timeInFive.getSeconds() + 5);
+
+  while (!workbook || new Date() > timeInFive) {
+    spaceData = await getSpace({
+      workflow: WorkflowType.Embedded,
+      spaceId: (existingSpace?.flatfileData as unknown as FlatfileSpaceData).id,
+    });
+
+    workbook = await getWorkbook({
+      workflow: WorkflowType.Embedded,
+      workbookId: spaceData?.primaryWorkbookId!,
+    });
+  }
+
+  if (!workbook) {
+    console.log("Unable to get workbook");
+    throw "Unable to get workbook";
+  }
+
+  const workbookConfig = {
+    name: workbook?.name || "HCM.show Embedded Portal",
+    sheets:
+      workbook?.sheets?.map((s) => {
+        return {
+          name: s.name,
+          slug: s.config?.slug,
+          fields: s.config?.fields,
+        };
+      }) || [],
+    actions: workbook?.actions || [],
```

Comment:
Do we need all these `?` and `[]`?

Reply:
Cleaned it up. Sheets on workbook is typed as a optional prop. 
![CleanShot 2023-06-14 at 13 54 43](https://github.com/FlatFilers/HCMShow/assets/12041898/fbbde1b5-f51d-44ff-8706-239f9f6933f3)